client/src/extension.ts

```diff
@@ -213,7 +211,23 @@ export async function activate (context: vscode.ExtensionContext): Promise<void>
 
   logger.info('Congratulations, your extension "BitBake" is now active!')
 
-  void vscode.commands.executeCommand('bitbake.rescan-project')
+  // Update the scan result stored in the global state
+  bitBakeProjectScanner.onChange.on('scanReady', (bitbakeScanResult: BitbakeScanResult) => {
+    void context.workspaceState.update('bitbake.ScanResult', bitbakeScanResult).then(() => {
+      logger.debug('BitBake scan result saved to global state')
+    })
+  })
+
+  const lastBitbakeScanResult: BitbakeScanResult | undefined = context.workspaceState.get('bitbake.ScanResult')
+
+  if (lastBitbakeScanResult === undefined) {

```

Comment:
That's cool not to have the scan starting when we already have the previous version. Things get ready much faster.

However, I wonder, should we start the scan nevertheless in case some changes happened outside of VS Code? (would there be any benefit in keeping a cache if we do so?)

Reply:
I think the tradeoff is good. Rarely will users modify the project outside VSCode. They can easily start a scan manually through the buttons in that case. We could explain this in the Troubleshooting.md tradeoffs.

Reply:
We could have a setting that lets users control this behaviour to cache the result or not. @deribaucourt What do you think about the changes that happened outside the vscode?

Reply:
My comment on this disappeared! I'm fine with not handling changes outside of bitbake in any way. Rarely will users have them. If they do, they can easily launch a scan manually form the UI. We can document this in troubleshooting.md if you're really worried IMO.

Reply:
sounds good to me